src/core/server/status/types.ts

```diff
@@ -134,9 +135,40 @@ export interface StatusServiceSetup {
    * only depend on the statuses of {@link StatusServiceSetup.core$ | Core} or their dependencies.
    */
   overall$: Observable<ServiceStatus>;
+
+  /**
+   * Allows a plugin to specify a custom status dependent on its own criteria.
+   * Completely overrides the default inherited status.
+   *
+   * @remarks
+   * See the {@link StatusServiceSetup.derivedStatus$} API for leveraging the default status
+   * calculation that is provided by Core.
+   */
+  set(status$: Observable<ServiceStatus>): void;
+
+  /**
+   * Current status for all dependencies of the current plugin.
+   * Each key of the `Record` is a plugin id.
+   */
+  plugins$: Observable<Record<string, ServiceStatus>>;
+
+  /**
+   * The status of this plugin as derived from its dependencies.
```

Comment:
shouldn't we add a warning that `derivedStatus$` might emit several times before the final status is calculated?

Reply:
Not a bad idea 👍 , though for the default case this should not happen.